src/Acuminator/Acuminator.Analyzers/Refactorings/EventHandlerSignature/EventHandlerSignatureRefactoring.cs

```diff
@@ -0,0 +1,160 @@
+					}
+				}
+			}
+		}
+
+		private bool IsSuitableForRefactoring(IMethodSymbol symbol, PXContext pxContext)
+		{
+			return symbol?.ContainingType?.OriginalDefinition != null
+			       && symbol.ContainingType.OriginalDefinition.IsPXGraphOrExtension(pxContext)
+			       && symbol.Parameters.Length <= 2
+			       && !symbol.IsOverride
+				   && !symbol.GetAttributes().Any(
+					   attr => attr.AttributeClass != null && attr.AttributeClass.Equals(pxContext.AttributeTypes.PXOverrideAttribute));
+		}
+
+		private Task<Document> ChangeSignatureAsync(CodeRefactoringContext context, 
+			SyntaxNode root, SemanticModel semanticModel,
+			MethodDeclarationSyntax methodDeclaration, IMethodSymbol methodSymbol,
+			EventType eventType, INamedTypeSymbol genericArgsSymbol,
+			string dacName, string fieldName,
+			CancellationToken cancellationToken)
+		{
+			cancellationToken.ThrowIfCancellationRequested();
+
+			var newParameters = methodDeclaration.ParameterList.Parameters;
+			var cacheParameterSymbol = methodSymbol.Parameters[0];
+			ParameterSyntax argsParameter;
+			SyntaxToken parameterName;
+
+			if (eventType == EventType.CacheAttached)
+			{
+				parameterName = Identifier(ArgsParameterName);
+				argsParameter = newParameters[0];
+			}
+			else
+			{
+				newParameters = newParameters.RemoveAt(0); // "PXCache sender" parameter
+				argsParameter = newParameters[0];
+				parameterName = argsParameter.Identifier;
+			}
+
+			newParameters = newParameters.Replace(argsParameter, 
+				CreateArgsParameter(genericArgsSymbol, parameterName, dacName, fieldName));
+
+			var newMethodDeclaration = 
+				ReplaceParameterUsages(methodDeclaration, cacheParameterSymbol,
+					MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, 
+						IdentifierName(parameterName), IdentifierName("Cache")), semanticModel)
+				.WithIdentifier(Identifier(EventHandlerMethodName))
+				.WithParameterList(methodDeclaration.ParameterList.WithParameters(newParameters));
+
+			root = root.ReplaceNode(methodDeclaration, newMethodDeclaration);
+
+			return Task.FromResult(context.Document.WithSyntaxRoot(root));
+		}
+
+		private ParameterSyntax CreateArgsParameter(INamedTypeSymbol genericArgsSymbol, SyntaxToken parameterName, 
+			string dacName, string fieldName)
+		{
+			TypeSyntax identifier;
+
+			if (String.IsNullOrEmpty(fieldName))
+				identifier = IdentifierName(dacName);
+			else
+				identifier = QualifiedName(IdentifierName(dacName), IdentifierName(fieldName));
+
+			return Parameter(parameterName)
+				.WithType(
+					QualifiedName(
+						IdentifierName(genericArgsSymbol.ContainingType.Name),
+						GenericName(
+								Identifier(genericArgsSymbol.Name))
+							.WithTypeArgumentList(
+								TypeArgumentList(
+									SingletonSeparatedList<TypeSyntax>(
+										identifier)))));
+		}
+
+		private (string dacName, string fieldName) ParseMethodName(string methodName, EventType eventType)
```

Comment:
@dropsonic I didn't find much info about naming conventions about named tuples but those that I found suggest to use Pascal Case for returned tuples since it makes the elements of the returned object to look like properties which is consistent with .Net naming conventions:
https://www.infoq.com/articles/Patterns-Practices-CSharp-7

Reply:
@SENya1990 ok

Reply:
@SENya1990 official blon on MSDN uses another naming convention:
https://blogs.msdn.microsoft.com/mazhou/2017/05/26/c-7-series-part-1-value-tuples/

Reply:
but documentation uses PascalCase: https://docs.microsoft.com/en-us/dotnet/csharp/tuples

Reply:
@dropsonic the official from your first reference uses it too for the returned tuples. The returned tuples has Pascal Case names and in local variables they use Camel case. I think this depends on the context - where you use tuple and how. It's still not very clear documented(